vault/ha.go

```diff
@@ -511,13 +506,18 @@ func (c *Core) waitForLeadership(doneCh, manualStepDownCh, stopCh chan struct{})
 		}
 
 		if releaseHALock {
-			if err := c.clearLeader(uuid); err != nil {
+			if err := c.clearLeader(ctx, uuid); err != nil {
 				c.logger.Error("clearing leader advertisement failed", "error", err)
 			}
 			c.heldHALock.Unlock()
 			c.heldHALock = nil
 		}
 
+		// Tell any requests that know about this to stop
+		if c.activeContextCancelFunc != nil {
+			c.activeContextCancelFunc()
```

Comment:
Could you move this back where it was, it's designed to stop all pending requests prior to grabbing the lock.

Reply:
I think all this PR needs to be is a single line changing c.activeContext in clearLeader to context.Background().

Reply:
@briankassouf moved it back up - sorry I misunderstood.

@jefferai I like having `ctx` as the first argument to `clearLeader`, since that's "the Go way", but I updated it to use its own `context.Background()` instantiated just before the call.